code/modules/mob/living/simple_animal/bot_swarm/swarmer.dm

```diff
@@ -0,0 +1,588 @@
+
+/obj/machinery/telecomms/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>This communications relay should be preserved, it will be a useful resource to our masters in the future. Aborting.</span>"
+
+/obj/machinery/message_server/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>This communications relay should be preserved, it will be a useful resource to our masters in the future. Aborting.</span>"
+
+/obj/machinery/power/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>Disrupting the power grid would bring no benefit to us. Aborting.</span>"
+
+/obj/machinery/gateway/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>This bluespace source will be important to us later. Aborting.</span>"
+
+/turf/simulated/wall/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	for(var/turf/T in range(1, src))
+		if(istype(T, /turf/space) || istype(T.loc, /area/space))
+			S << "<span class='warning'>Destroying this object has the potential to cause a hull breach. Aborting.</span>"
+			return
+	..()
+
+/obj/structure/window/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	for(var/turf/T in range(1, src))
+		if(istype(T, /turf/space) || istype(T.loc, /area/space))
+			S << "<span class='warning'>Destroying this object has the potential to cause a hull breach. Aborting.</span>"
+			return
+	..()
+
+/obj/item/stack/cable_coil/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)//Wiring would be too effective as a resource
+	S << "<span class='warning'>This object does not contain enough materials to work with.</span>"
+
+/obj/machinery/porta_turret/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>Attempting to dismantle this machine would result in an immediate counterattack. Aborting.</span>"
+
+/mob/living/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S.DisperseTarget(src)
+
+/mob/living/simple_animal/slime/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>This biological resource is somehow resisting our bluespace transceiver. Aborting.</span>"
+
+////END CTRL CLICK FOR SWARMERS////
+
+/mob/living/simple_animal/hostile/swarmer/proc/Fabricate(var/atom/fabrication_object,var/fabrication_cost = 0)
+	if(!isturf(loc))
+		src << "<span class='warning'>This is not a suitable location for fabrication. We need more space.</span>"
+	if(resources >= fabrication_cost)
+		resources -= fabrication_cost
+	else
+		src << "<span class='warning'>You do not have the necessary resources to fabricate this object.</span>"
+		return 0
+	new fabrication_object(loc)
+	return 1
+
+/mob/living/simple_animal/hostile/swarmer/proc/Integrate(var/obj/item/target)
+	if(resources >= 100)
+		src << "<span class='warning'>We cannot hold more materials!</span>"
+		return
+	if((target.materials[MAT_METAL]) || (target.materials[MAT_GLASS]))
```

Comment:
Should probably check if the materials list is defined and contains these materials rather than try to access the index of something that might not exist.

```
//Make sure the materials list has at least one entry
if(target.materials && target.materials.len)
   //Check if any entries are either MAT_METAL or MAT_GLASS
   if((MAT_METAL in target.materials) || (MAT_GLASS in target.materials))
      //code to increase resources and such goes here
```